pkg/watermark/store/jetstream/kv_watch.go

```diff
@@ -35,45 +34,62 @@ type jetStreamWatch struct {
 	kvBucketName string
 	conn         *jsclient.NatsConn
 	js           *jsclient.JetStreamContext
+	kvWatchers   []*KeyWatcher
 	log          *zap.SugaredLogger
 }
 
 var _ store.WatermarkKVWatcher = (*jetStreamWatch)(nil)
 
 // NewKVJetStreamKVWatch returns KVJetStreamWatch specific to JetStream which implements the WatermarkKVWatcher interface.
 func NewKVJetStreamKVWatch(ctx context.Context, pipelineName string, kvBucketName string, client jsclient.JetStreamClient, opts ...JSKVWatcherOption) (store.WatermarkKVWatcher, error) {
-	var err error
-	conn, err := client.Connect(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get nats connection, %w", err)
+	jsw := &jetStreamWatch{
+		pipelineName: pipelineName,
+		kvBucketName: kvBucketName,
+		kvWatchers:   make([]*KeyWatcher, 0),
+		log:          logging.FromContext(ctx).With("pipeline", pipelineName).With("kvBucketName", kvBucketName),
 	}
 
-	js, err := conn.JetStream()
-	if err != nil {
-		if !conn.IsClosed() {
+	connectAndWatch := func() (*jsclient.NatsConn, *jsclient.JetStreamContext, error) {
+		conn, err := client.Connect(ctx, jsclient.ReconnectHandler(func(c *jsclient.NatsConn) {
```

Comment:
Can this also handle the nats native auto reconnect caused problem?

Reply:
After several attempts, I noticed watcher stops working with nats native auto-reconnect as well and therefore modified the logic to disable it.